File: hono-api/src/auth/auth.schema.ts
import { z } from "zod";

export const registerSchema = z.object({
    'firstName':z.string().min(3),
    'lastName':z.string().min(4),
    'email':z.string(),
    'password':z.string().min(3),
    'role':z.string().optional()
})
export const loginSchema=z.object({
    'email':z.string(),
    'password':z.string()
})

File: flask-api/app/routes/schema/auth_schema.py
from flask_restx import Namespace,fields

auth_ns=Namespace('auth',description='auth operations')

register_data=auth_ns.model(
    'register',
    {
    'first_name':fields.String(required=True,description='user first name'),
    'last_name':fields.String(required=True,description='user last name'),
    'email':fields.String(required=True,description='user email'),
    'password':fields.String(required=True,description='user email'),
    'role':fields.String(default='user',description='user role')
    }
)

user_output = auth_ns.model(
    "UserOutput",
    {

    'first_name':fields.String(required=True,description='user first name'),
    'last_name':fields.String(required=True,description='user last name'),
        "email": fields.String(description="User's email"),
    },
)

login_data=auth_ns.model('login',{
     'email':fields.String(required=True,description='user email'),
    'password':fields.String(required=True,description='user email'),
    
})